wallet-new/test/unit/Test/Spec/Addresses.hs

```diff
@@ -408,55 +408,50 @@ spec = describe "Addresses" $ do
-                                -- and each account has an initial address (but not the initial
-                                -- account thus the -1)
-                                length (wrData wr) `shouldBe` expectedNumber 3 4
+                                length (wrData wr) `shouldBe` min 40 total
                             _        -> fail ("Got " ++ show res)
 
             prop "is deterministic" $ withMaxSuccess 20 $ do
                 monadicIO $
-                    withAddressesFixtures 3 8 $ \_ layer _ _ -> do
-                        let (expectedTotal :: Int) = expectedNumber 3 8
-                        let pp = PaginationParams (Page 1) (PerPage 40)
-                        let pp1 = PaginationParams (Page 1) (PerPage (quot expectedTotal 3 + 1))
-                        let pp2 = PaginationParams (Page 2) (PerPage (quot expectedTotal 3 + 1))
-                        let pp3 = PaginationParams (Page 3) (PerPage (quot expectedTotal 3 + 1))
-                        res <- runExceptT $ runHandler' $ do
-                            Handlers.listAddresses layer (RequestParams pp)
-                        res' <- runExceptT $ runHandler' $ do
-                            Handlers.listAddresses layer (RequestParams pp)
-                        res1 <- runExceptT $ runHandler' $ do
-                            Handlers.listAddresses layer (RequestParams pp1)
-                        res1' <- runExceptT $ runHandler' $ do
-                            Handlers.listAddresses layer (RequestParams pp1)
-                        res2 <- runExceptT $ runHandler' $ do
-                            Handlers.listAddresses layer (RequestParams pp2)
-                        res2' <- runExceptT $ runHandler' $ do
-                            Handlers.listAddresses layer (RequestParams pp2)
-                        res3 <- runExceptT $ runHandler' $ do
-                            Handlers.listAddresses layer (RequestParams pp3)
-                        res3' <- runExceptT $ runHandler' $ do
-                            Handlers.listAddresses layer (RequestParams pp3)
-                        res `shouldBe` res'
-                        res1 `shouldBe` res1'
-                        res2 `shouldBe` res2'
-                        res3 `shouldBe` res3'
+                    withAddressesFixtures 3 8 $ \_ layer _ (_, expectedTotal) -> do
+                        let ppSplit = quot expectedTotal 3 + 1
+                            pp = PaginationParams (Page 1) (PerPage 40)
+                            pp1 = PaginationParams (Page 1) (PerPage ppSplit)
+                            pp2 = PaginationParams (Page 2) (PerPage ppSplit)
+                            pp3 = PaginationParams (Page 3) (PerPage ppSplit)
+                            mkRequest mypp = Handlers.listAddresses layer (RequestParams mypp)
+                        _ <- runExceptT $ runHandler' $ do
```

Comment:
Nice, much better? I was thinking that in theory we could simplify this even further, can't we? In principle, we could do something like:

```
forM [(1,40), (2, ppSplit), (2, ppSplit), (3, ppSplit)] $ \(page, perPage) -> do
    pp = PaginationParams (Page page, PerPage perPage)
    (r1, r2) <- runExceptT $ runHandler' $ do
                      (,) <$> mkRequest pp <*> mkRequest pp
     r1 `shouldBe` r2
```

What do you think?